src/s3ql/backends/gdrive.py

```diff
@@ -0,0 +1,345 @@
+                    b64 += properties[self._chunk_property(k, i)]
+                metadata[k] = base64.b64decode(b64)
+            elif v.isdigit():
+                metadata[k] = int(v)
+            else:
+                metadata[k] = v
+        log.debug('{0}: metadata {1}'.format(f['name'], metadata))
+        return metadata
+
+    @property
+    @copy_ancestor_docstring
+    def has_native_rename(self):
+        return False
+
+    def __str__(self):
+        return 'gdrive folder %s@%s' % (self.login, self.prefix)
+
+    @copy_ancestor_docstring
+    def is_temp_failure(self, exc): #IGNORE:W0613
+        return False
+
+    @copy_ancestor_docstring
+    def lookup(self, key):
+        log.debug("lookup {0}".format(key))
+        f = self._lookup_file(key, self.folder)
+        if f is None:
+            raise NoSuchObject(key)
+        return self._decode_metadata(f)
+
+    @copy_ancestor_docstring
+    def get_size(self, key):
+        log.debug("get_size {0}".format(key))
+        f = self._lookup_file(key, self.folder)
+        if f is None:
+            raise NoSuchObject(key)
+        return int(f.get('size', u'0'))
+
+    @copy_ancestor_docstring
+    def open_read(self, key):
+        log.debug("open_read {0}".format(key))
+        f = self._lookup_file(key, self.folder)
+        if f is None:
+            raise NoSuchObject(key)
+        return ObjectR(self.service, f, self._decode_metadata(f))
+
+    @copy_ancestor_docstring
+    def open_write(self, key, metadata=None, is_compressed=False):
+        log.debug("open_write {0}".format(key))
+        if metadata is None:
+            metadata = dict()
+        elif not isinstance(metadata, dict):
+            raise TypeError('*metadata*: expected dict or None, got %s' % type(metadata))
+
+        body = {
+            'name': key,
+            'mimeType': Backend.MIME_TYPE_BINARY,
+            'parents': [ self.folder['id'] ],
+            'properties': self._encode_metadata(metadata),
+        }
+        log.debug("metadata: {0}, body: {1}".format(metadata, body))
+        f = self._create_file(body)
+        self._delete_other(f)
+        return ObjectW(self.service, f)
+
+    @copy_ancestor_docstring
+    def clear(self):
+        log.debug("clear")
+        for f in self._list_files(self.folder, "id"):
+            self._delete_file(f)
+
+    @copy_ancestor_docstring
+    def contains(self, key):
+        log.debug("contains {0}".format(key))
+        f = self._lookup_file(key, self.folder)
+        return f is not None
+
+    @copy_ancestor_docstring
+    def delete(self, key, force=False):
+        log.debug("delete {0}".format(key))
+        f = self._lookup_file(key, self.folder)
+        if f is not None:
+            self._delete_file(f)
+        elif not force:
+            raise NoSuchObject(key)
+
+    @copy_ancestor_docstring
+    def list(self, prefix=''):
+        log.debug("list {0}".format(prefix))
+        for f in self._list_files(self.folder, "id, name"):
```

Comment:
Is there no way to tell Google Drive to list only files starting with the prefix?